### ISSUE TITLE
DataBaseChangeListener not working as expected

### ISSUE THREAD

**Issue by Masquerade0097** [fossasia/open-event-orga-app#1079]
### Bug Report    

**Actual Behaviour**

DataBaseChangeListener is not working as expected. The UI is not getting updated for following changes.
1. When a Track is updated.
2. When a Session is updated.
3. When a Faq is created.
4. When Sponsor is updated.

**Expected Behaviour**

The UI should be updated automatically with changes. The user should not have to refresh explicitly.

**Android version and Phone Model**

Nougat and Moto G4 Plus

**Comment by srv-twry** [comment]
@Masquerade0097 Should I take it if you aren't working on this ?

**Comment by srv-twry** [comment]
The change listener is working fine. I have found the problem. There are two different cases of this issue:
1. When the create/update fragment is a `BaseBottomSheetFragment`. For eg. Create/Update Track.
This is easy to fix.

2. When the create/update fragment is a complete fragment i.e. the list fragment is hidden. In this case the database change listener is actually detached since `onStop` is called for the list fragment. As a result the listener isn't triggered at all. I can't figure out a way to fix this case except migrating to MVVM. 

@iamareebjamal Can you please help in the second case ?

**Comment by iamareebjamal** [comment]
But when the fragment is re-attached, onStart is called and presenter must load the new information?

**Comment by srv-twry** [comment]
Yes, the onStart is called but since the presenter already has information as an in-memory variable, data is returned from it only rather than reload from db.

```Java
public void loadSessions(boolean forceReload) {
        getSessionSource(forceReload)
            .compose(dispose(getDisposable()))
            .compose(progressiveErroneousRefresh(getView(), forceReload))
            .toList()
            .compose(emptiable(getView(), sessions))
            .subscribe(Logger::logSuccess, Logger::logError);
    }

    private Observable<Session> getSessionSource(boolean forceReload) {
        if (!forceReload && !sessions.isEmpty() && isRotated())
            return Observable.fromIterable(sessions);
        else {
            return sessionRepository.getSessions(getId(), forceReload);
        }
    }

```

When the `onStart` is called, it calls `loadSessions(false)` which calls the `getSessionSource(false)`.
Now this if statement is true and hence data is loaded from the in-memory sessions list.
```Java
 if (!forceReload && !sessions.isEmpty() && isRotated())
            return Observable.fromIterable(sessions);
```

We need to somehow tell it to load from the db and not from the in-memory values.

**Comment by iamareebjamal** [comment]
OK, so we need to reload from db always, I don't think there's a point storing in memory in presenters

We'll be changing to MVVM so that'll happen automatically later

**Comment by Masquerade0097** [comment]
@iamareebjamal Since, DatabaseChangeListener is working fine and it will be resolved after migrating to MVVM therefore, should it be closed to avoid overlapping issues as #562 ?
